From 593b6466364de895e906de2457979d8588c73f92 Mon Sep 17 00:00:00 2001
From: mkosmach <mkosmach@gmail.com>
Date: Wed, 17 Mar 2021 21:02:59 +0300
Subject: [PATCH] Fix for multicast entries in "ip -6 route show" output in
 salt/modules/network.py

Patrial fix (only for multicast entries for now) for saltstack/salt#58377
---
 salt/modules/network.py | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/salt/modules/network.py b/salt/modules/network.py
index 9280a0f854fcb..b5eb746044da2 100644
--- a/salt/modules/network.py
+++ b/salt/modules/network.py
@@ -605,6 +605,23 @@ def _ip_route_linux():
                     "interface": ip_interface,
                 }
             )
+
+        elif comps[0] == "multicast":
+            ip_interface = ""
+            if comps[2] == "dev":
+                ip_interface = comps[3]
+
+            ret.append(
+                {
+                    "addr_family": "inet6",
+                    "destination": comps[1],
+                    "gateway": comps[4],
+                    "netmask": "",
+                    "flags": "U",
+                    "interface": ip_interface,
+                }
+            )
+
         else:
             address_mask = convert_cidr(comps[0])
             ip_interface = ""
